src/Workspaces/Core/Portable/Workspace/Solution/AdditionalDocumentState.cs

```diff
@@ -39,6 +39,14 @@ protected override TextDocumentState UpdateAttributes(DocumentInfo.DocumentAttri
             TextAndVersionSource,
             LoadTextOptions);
 
+    protected override TextDocumentState UpdateDocumentServiceProvider(IDocumentServiceProvider? newProvider)
```

Comment:
when/how does this change?

Reply:
A different provider can be passed via `DocumentInfo`